backend/internal_assembler/internal_assembler.ml

```diff
@@ -0,0 +1,255 @@
+module SectionName = X86_proc.SectionName
+module StringMap = X86_binary_emitter.StringMap
+module StringTbl = X86_binary_emitter.StringTbl
+module SectionMap = Map.Make (SectionName)
+module SectionTbl = Hashtbl.Make (SectionName)
```

Comment:
You could move these directly into `Section_name` (which should be capitalised like that, not `SectionName`).  This would give e.g. `Section_name.Map` which is clearer.

Reply:
The `.mli` will need something like the following:
```
include Map.S with type key := t
```